src/DataProtection/DataProtection/src/Cng/GcmAuthenticatedEncryptor.cs

```diff
@@ -151,7 +152,18 @@ protected override byte[] DecryptImpl(byte* pbCiphertext, uint cbCiphertext, byt
                         cbDerivedKey: _symmetricAlgorithmSubkeyLengthInBytes);
 
                     // Perform the decryption operation
-
+#if NETCOREAPP
+                    unsafe
+                    {
+                        var nonce = new Span<byte>(pbNonce, (int)NONCE_SIZE_IN_BYTES);
+                        var key = new Span<byte>(pbSymmetricDecryptionSubkey, (int)_symmetricAlgorithmSubkeyLengthInBytes);
+                        var tag = new Span<byte>(pbAuthTag, (int)TAG_SIZE_IN_BYTES);
+                        var plaintext = new Span<byte>(retVal);
+                        var encrypted = new Span<byte>(pbEncryptedData, (int)cbPlaintext);
+                        using var aes = new AesGcm(key);
```

Comment:
Does this dispose at the end of the block or at the end of the method (99% sure the answer is block, but want to make sure. It might be better to use the old using syntax to be very explicit)

Reply:
Yeah from the docs I believe its the block (once it goes out of scope) https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement 

I can change it to have braces if you feel strongly about it, or we can just start training ourselves to read these new usings now :)

Reply:
I'm fine training myself, just bring it up because it's "security" sensitive code, in any other case I wouldn't have blinked.